### PR TITLE
Fix #5703: Improve error message when events fail to load

### LINKED ISSUE #5703: Improve handling of event 404s
Per an internal discussion, instead of getting this rather opaque error message:

screen shot 2017-07-10 at 3 49 09 pm

It'd be nice to see something a bit more informative:

better-issue-404

@getsentry/product

### PR DESCRIPTION
**Pull request by billyvg**
In `<GroupEventDetails>`, swapped out `<LoadingError>` with a new `<DetailedError>`.

image

Fixes #5703

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/components/errors/groupEventDetailsLoadingError.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const reasons = [
+    t('The events are still processing and are on their way'),
+    t('The events have been deleted'),
+    t('There is an internal systems error or active issue')
+  ];
+
+  return (
+    <DetailedError
+      onRetry={onRetry}
+      heading="Sorry, the events for this issue could not be found."
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/errors/groupEventDetailsLoadingError.jsx]: this should be translated too

File: src/sentry/static/sentry/app/components/errors/groupEventDetailsLoadingError.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      message={
+        <div>
+          <p>This could be due to a handful of reasons:</p>
```
Comment (macqueen) [inline src/sentry/static/sentry/app/components/errors/groupEventDetailsLoadingError.jsx]: and this

**Review by macqueen — COMMENTED** [review]
looks awesome! one question: why is the pagination/header showing in the screenshot? it seems like that shouldn't be available

File: src/sentry/static/sentry/app/components/errors/groupEventDetailsLoadingError.jsx (hunk 2 above)
Comment (billyvg) [inline src/sentry/static/sentry/app/components/errors/groupEventDetailsLoadingError.jsx]: Oops, thanks!

**Comment by billyvg** [comment]
Ah the reason the toolbar is showing is because I was only setting the `error` state in `<GroupEventDetails>` and the toolbar is displayed as long as the `event` data is in state.

File: src/sentry/static/sentry/less/errors.less [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.detailed-error {
+  display: flex;
+  justify-content: center;
+  padding: 18px;
```
Comment (ckj) [inline src/sentry/static/sentry/less/errors.less]: After playing around with this locally, I think this needs bumped up to something like 40px. It's a bit tight for my taste as it is.

**Review by ckj — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/less/errors.less (hunk 3 above)
Comment (ckj) [inline src/sentry/static/sentry/less/errors.less]: NM! Looks like you covered it elsewhere in your last commit.

**Review by ckj — APPROVED** [review]
🤘

**Pull request merged**